samply/src/linux_shared/mod.rs

```diff
@@ -252,6 +255,111 @@ where
         }
     }
 
+    fn process_perf_map_line(line: &str) -> Option<(u64, u64, &str)> {
+        let mut split = line.splitn(3, ' ');
+        let addr = split.next()?;
+        let len = split.next()?;
+        let symbol_name = split.next()?;
```

Comment:
I've changed this to use `splitn` so that we wouldn't need to glue together the last piece with spaces again. I hope perf.map files always use just a single space as a separator.

Reply:
Oh, yeah, [the "spec"](https://github.com/torvalds/linux/blob/master/tools/perf/Documentation/jit-interface.txt) says:

> Each line has the following format, fields separated with spaces:

Reply:
Sweet, TIL `splitn`.